src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
```

Comment:
Admittedly I haven't spent much mental energy thinking about this, but...what are you thinking is the advantage of introducing function maps in this code?

I'm not against it, just curious.